apps/emqx_rule_engine/src/emqx_rule_actions.erl

```diff
@@ -133,6 +144,7 @@ on_action_inspect(Selected, Envs) ->
 on_action_create_republish(Id, Params = #{
         <<"target_topic">> := TargetTopic,
         <<"target_qos">> := TargetQoS,
+        <<"target_retain">> := TargetRetain,
         <<"payload_tmpl">> := PayloadTmpl
        }) ->
```

Comment:
And here we get the optional option `target_retain` and set a default value, such as:

```erlang
TargetRetain = case maps:get(<<"target_retain">>, TargetRetain, 0) of
  1 -> true;
  _ -> false
end,
```

Reply:
The variable `TargetRetain` here will be passed to `on_action_republish/2`, included in the `?BINDING_KEYS` of the second arguments `Envs`. This work is done by emqx-rule-engine automatically.